CMakeLists.txt

```diff
@@ -24,174 +39,108 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
   list(APPEND boost_components zlib)
 endif()
 find_package(Boost 1.51 REQUIRED COMPONENTS ${boost_components})
-include_directories(${Boost_INCLUDE_DIRS})
 
 find_package(CURL REQUIRED)
-include_directories(${CURL_INCLUDE_DIR})
-
-find_package(LZ4 REQUIRED)
-include_directories(${LZ4_INCLUDE_DIR})
+add_library(CURL::CURL INTERFACE IMPORTED)
+set_target_properties(CURL::CURL PROPERTIES
+  INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}"
+  INTERFACE_INCLUDE_DIRECTORIES "${CURL_INCLUDE_DIR}"
+  INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
 
 find_package(Lua REQUIRED)
-include_directories(${LUA_INCLUDE_DIR})
-
-find_package(Protobuf REQUIRED)
-include_directories(${PROTOBUF_INCLUDE_DIRS})
+add_library(Lua::Lua INTERFACE IMPORTED)
+set_target_properties(Lua::Lua PROPERTIES
+  INTERFACE_LINK_LIBRARIES "${LUA_LIBRARIES}"
+  INTERFACE_INCLUDE_DIRECTORIES "${LUA_INCLUDE_DIR}")
 
+find_package(ZLIB REQUIRED)
+find_package(LZ4 REQUIRED)
 find_package(SQLite3 REQUIRED)
-
 find_package(SpatiaLite)
+find_package(GEOS)
+
+add_library(libprime_server INTERFACE IMPORTED)
```

Comment:
does this mean we get `prime_server` no matter what optional stuff we were asking for? i would only add prime_server `if(ENABLE_SERVICES)` if its possible

Reply:
:+1: removed `ENABLE_HTTP` option and changed  `libprime_server` to a required dependency  if `ENABLE_SERVICES` is On